build.gradle

```diff
@@ -107,23 +134,47 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String findNewestEapBuildNumber() {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href~=('+ project.intellijSnapshotsUrl + 'com/jetbrains/intellij/idea/ideaIC/)\\d+\\.\\d+\\.\\d+][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("(?<=ideaIC-)\\d+\\.\\d+\\.\\d+(?=-EAP-SNAPSHOT.pom)")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    println matcher.group()
+    return matcher.group()
   }
+  return null
+}
 
-  @TaskAction
-  void print() {
-    println(String.join("\n", project.resolveIntelliJVersions(versionKey)))
+String checkForEapWithBuildNumberHigherThan(String latestEapBuildNumber) {
+  String foundBuildNumber = project.findNewestEapBuildNumber()
+  if (foundBuildNumber) {
+    if (VersionNumber.parse(foundBuildNumber) > VersionNumber.parse(latestEapBuildNumber)) {
+      return foundBuildNumber
+    }
   }
+  return null
 }
 
-tasks.register('resolveIntelliJVersions', ResolveIntelliJVersions)
+task updateEapBuildNumber {
```

Comment:
Is this task now executed automagically every time `./gradlew` is run? 😯 at least that's what I'm observing locally 🤔